iothub/device/src/ClientPropertyCollection.cs

```diff
@@ -442,6 +480,12 @@ private void AddInternal(IDictionary<string, object> properties, string componen
 
                 foreach (KeyValuePair<string, object> entry in properties)
                 {
+                    // A null property key is not allowed. Throw an ArgumentNullException.
+                    if (entry.Key == null)
```

Comment:
As an aside, a dictionary will not allow you to have a null value as a key. Might be overkill to check for this unless we're thinking that someone will violate the IDictionary<> Add method.

Reply:
It will likely save us from some headache since we don't want to be in a position where we would try to serialize a null key